plugins/modules/foreman_provisioning_template.py

```diff
@@ -250,25 +250,13 @@ def find_template_kind(module, entity_dict):
     return entity_dict
 
 
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'audit_comment': 'audit_comment',
-    'kind': 'template_kind_id',
-    'locations': 'location_ids',
-    'locked': 'locked',
-    'name': 'name',
-    'organizations': 'organization_ids',
-    'operatingsystems': 'operatingsystem_ids',
-    'snippet': 'snippet',
-    'template': 'template',
-}
-# Missing parameters:
-# default
-
-
 def main():
     module = ForemanEntityApypieAnsibleModule(
         argument_spec=dict(
+            file_name=dict(type='path'),
+            state=dict(default='present', choices=['absent', 'present_with_defaults', 'present']),
+        ),
+        entity_spec=dict(
             audit_comment=dict(),
             kind=dict(choices=[
```

Comment:
Possible future improvement: can we know this from apypie? We probably can't document it, but I think apidoc.json should have this info at least for validation.

Reply:
chicken-n-egg sandwich, the validation of the choices happens *before* we initiated the connection to the server, so there is no apidoc available yet.

Reply:
But what if Foreman adds additional options? Either via a new version or a plugin.

Reply:
Then this list is sadly outdated.

Reply:
Also we have to keep the docs up to date with this manually. :smiling_imp: